HelpSource/Guides/UserFAQ.schelp

```diff
@@ -0,0 +1,418 @@
+    // ...
+});
+::
+
+subsection:: Why does it have to be a literal array?
+
+The reason comes from the process of building a SynthDef:
+
+1. First, look at the function arguments to figure out what the Control inputs should be.
+2. Then create Control units (usually just one, if they're all normal arguments without prefixes or special rates). Each channel is represented by an code::OutputProxy::.
+3. Then run the SynthDef function, passing the output proxies to the arguments.
+4. Then sort the UGens into the right order, etc. etc.
+
+To do steps \#1 and \#2, the SynthDef builder has to know the size of an array argument strong::before:: running the function. That's possible only if it's a literal array: code::\#[1, 2, 3, 4, 5]::. Any other array notation creates the array strong::while running the function:: (step \#3). But then it's too late -- the SynthDef builder already created a non-array control channel for it!
+
+code::
+SynthDef(\notArray, { |a = (1..5)|
+    a.debug("a is");
+});
+::
+code::a is: an OutputProxy::
+
+code::
+SynthDef(\array, { |a = #[1, 2, 3, 4, 5]|
+   a.debug("a is");
+});
+::
+code::a is: [ an OutputProxy, an OutputProxy, an OutputProxy, an OutputProxy, an OutputProxy ]::
+
+(Note, if 'a' printed as [ 1, 2, 3, 4, 5 ], then you wouldn't be able to change the values in a Synth using code::.set::!)
+
+
+anchor::Server Issues::
+
+SECTION:: Server issues
+
+subsection:: How to trigger a function from the server
+
+The first and most important point: strong::Functions are client-side only::.
+The server doesn't know what functions are, doesn't understand them and has no way to execute them.
+strong::Only the client can execute a function::.
+
+Therefore, if you want a function to execute when something happens in the server, the only way is for the server to tell the client to take the action.
+
+The server can communicate messages back to the client using one of two unit generators: code::SendTrig:: and code::SendReply::.
+code::SendTrig:: is simpler and less flexible (it can send only a code::/tr:: message, and only one data value).
+code::SendReply:: allows you to name the message anything you like, and can send arrays with the message.
+We'll use SendReply here because of its greater flexibility.
+
+Within the language, you also need an object to receive the message and act on it. Usually this is code::OSCresponderNode:: or code::OSCpathResponder::.  In this example, code::OSCpathResponder:: filters messages not just on the name code::/bleep:: but also on the synth's ID. This way, you could have multiple triggering synths, with a different responder and a different action per synth.
+
+code::
+(
+a = {
+    var trig = Dust.kr(8),
+    decay = Decay2.kr(trig, 0.01, 0.1),
+    sig = SinOsc.ar(TExpRand.kr(200, 600, trig), 0, 0.1) * decay;
+    SendReply.kr(trig, '/bleep', trig);
+    sig ! 2
+}.play;
+
+o = OSCpathResponder(s.addr, ['/bleep', a.nodeID], { |time, thisResponder, msg|
```

Comment:
It would be great to replace the responder with OSCdef:
```supercollider
o = OSCdef(\bleepResponder, { |msg|
	msg.postln;
}, '/bleep', s.addr, argTemplate: [a.nodeID]);
```